src/main/java/com/teragrep/cfe_16/event/JsonEventImpl.java

```diff
@@ -58,24 +58,32 @@ public JsonEventImpl(final JsonNode jsonNode) {
 
     @Override
     public EventMessage asEvent() {
+        final EventMessage eventMessage;
         // Event field completely missing
         if (!this.asNode().has("event")) {
-            return new EventMessageStub();
+            eventMessage = new EventMessageStub();
```

Comment:
stub is immutable an therefore the same instance can and should be shared accross all stub returns, please change as such.